src/lambda_streams/Animals.java

```diff
@@ -1,62 +1,173 @@
+    {
         //lowercase the first letter, and uppercase the rest of the letters, using streams.  Also, depending on the value of the boolean flag 'mutate', mutate the original animals list, or perform your stream operations on a 'copy' of the animals list.  return the list out of hte function in both cases.
+        if(mutate)
+        {
+            animals = animaList.stream()
+                    .map(x -> x.substring(0, 1).toLowerCase() + x.substring(1).toUpperCase())
+                    .collect(Collectors.toList());
+        } else {
+            List<String> animaList3 = animaList;
+            return animaList3.stream()
+                    .map(x -> x.substring(0, 1).toLowerCase() + x.substring(1).toUpperCase())
+                    .collect(Collectors.toList());
+        }
         return animaList;
     }
 
-    static List<String> flipAnimals(boolean mutate) {
+    static List<String> flipAnimals(boolean mutate)
+    {
         //reverse the order of the animals in the animal list.  If the booleaen parameter is true, reverse the static animals array list by mutating the array.  if the mutate boolean is false, flip a 'copy' of the animals list, then return that list of flipped animals, WITHOUT mutating the static animals array. return the flipped list in both cases.
 
-        List<String> instructions = Arrays.asList("Flip","the","animals","list","."," ","Mutate","the","animals","only","if","flag","true");
-        return instructions;
+//        List<String> instructions = Arrays.asList("Flip","the","animals","list","."," ","Mutate","the","animals","only","if","flag","true");
+//        return instructions;
+        if(mutate)
+        {
+            Collections.reverse(animals);
+        } else {
+            List<String> animaList4 = animals;
+            Collections.reverse(animaList4);
+        }
+        return animals;
     }
 
-    static List<String> sortAnimals(boolean mutate) {
-    //sort the animals in alphabetical order.  If the booleaen parameter is true, mutating the animals list.  if the mutate boolean is false, sort a 'copy' of the animals list, then return that list of sorted animals, WITHOUT mutating the static animals array. return the sorted list in both cases.
-    return animals;
+    static List<String> sortAnimals(boolean mutate)
+    {
+        //sort the animals in alphabetical order.  If the booleaen parameter is true, mutating the animals list.  if the mutate boolean is false, sort a 'copy' of the animals list, then return that list of sorted animals, WITHOUT mutating the static animals array. return the sorted list in both cases.
+        if (mutate)
+        {
+            Collections.sort(animals);
+        } else {
+            List<String> animaList5 = animals;
+            Collections.sort(animaList5);
+        }
+        return animals;
     }
 
 
 }
+
+@FunctionalInterface
+interface LambdaListBool
+{
+    List<String> calc(List<String> list, boolean mutate);
+}
```

Comment:
This approach results in a .java file contain two top-level classes of different names.  Let's say someone knows there is a defect with the LambdaListBool interface, but doesn't know exactly where it is in the code.  With the interface "hidden" within this file, it might take them quite a while to find.  The preference is for every top-level Type to be in its own .java file.